### DISCUSSION TITLE
Enumerating implicit roles

### DISCUSSION THREAD

**Discussion by Bilge** [symfony/symfony#44927] [Q&A] [answered]
Given that explicit roles are those we define for `UserInterface::getRoles` and implicit roles are those we define in the `security.role_hierarchy` configuration: we can enumerate explicit roles by simply iterating over `UserInterface::getRoles`, but it seems we can only interact with implicit roles by querying specific ones using `AuthorizationCheckerInterface::isGranted`. How can we enumerate all the implicit roles also granted to a user through the role hierarchy?

**Comment by stof** [comment, marked as answer]
Use `RoleHierarchyInterface::getReachableRoleNames`

**Reply by Bilge** [reply]
Thanks!
